guard/src/commands/validate.rs

```diff
@@ -128,64 +148,106 @@ Note - When pointing the command to a directory, the directory may not contain a
 rules and data files. The directory being pointed to must contain only data files,
 or rules files.
 "#)
-            .arg(Arg::with_name(RULES.0).long(RULES.0).short(RULES.1).takes_value(true)
+            .arg(Arg::new(RULES.0)
+                .long(RULES.0)
+                .short(RULES.1)
+                .num_args(0..)
+                .action(ArgAction::Append)
                 .help("Provide a rules file or a directory of rules files. Supports passing multiple values by using this option repeatedly.\
                           \nExample:\n --rules rule1.guard --rules ./rules-dir1 --rules rule2.guard\
                           \nFor directory arguments such as `rules-dir1` above, scanning is only supported for files with following extensions: .guard, .ruleset")
-                .multiple(true)
                 .conflicts_with("payload"))
-            .arg(Arg::with_name(DATA.0).long(DATA.0).short(DATA.1).takes_value(true)
+            .arg(Arg::new(DATA.0)
+                .long(DATA.0)
+                .short(DATA.1)
+                .num_args(0..)
```

Comment:
AFAIU the documentation recommends against this approach and suggests to use either of the two `ArgAction::Append` or `num_args`. Is that not the case? Have you verified by running the command with multiple `data` and `rules` args and it worked?

[Reference](https://docs.rs/clap/latest/clap/struct.Arg.html#method.num_args)

> A common mistake is to define an option which allows multiple values and a positional argument.

> The problem is clap doesn’t know when to stop parsing values for “file”.

> A solution for the example above is to limit how many values with a maximum, or specific number, or to say [ArgAction::Append](https://docs.rs/clap/latest/clap/builder/enum.ArgAction.html#variant.Append) is ok, but multiple values are not.

Reply:
>  Is that not the case? Have you verified by running the command with multiple data and rules args and it worked?

we have e2e tests that already do this for us? 

it might be redundant to have both, will see about removing num_args

Reply:
furthermore without num_args(0..) the only way clap would allow us to have multiple values would be to either specify a delimiter or to keep passing either the rules/data/input-params flag everytime we add another value. 

This approach provides the same exact behaviour to the customer